From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 25 Jul 2022 02:42:00 +0200
Subject: [PATCH] Disable little-used image formats

Wezterm is not an image editor, it doesn't make much sense to support PNM,
DDS, TGA, OpenEXR, TIFF and similar rare image formats - it just adds more
bloat to the binaries.

diff --git a/term/Cargo.toml b/term/Cargo.toml
index 513a03efc..5b2bd6733 100644
--- a/term/Cargo.toml
+++ b/term/Cargo.toml
@@ -23,7 +23,7 @@ miniz_oxide = "0.4"
 finl_unicode = "1.2"
 hex = "0.4"
 # we want image 0.24.6 or later. see https://github.com/wez/wezterm/issues/3250
-image = {version="0.24", git="https://github.com/image-rs/image.git", rev="fe069785ae245a2c510fd724ef96da283b05a236"}
+image = { version = "0.24", git = "https://github.com/image-rs/image.git", rev = "fe069785ae245a2c510fd724ef96da283b05a236", default-features = false, features = ["gif", "jpeg", "ico", "png", "farbfeld", "jpeg_rayon"] }
 lazy_static = "1.4"
 log = "0.4"
 lru = "0.7"
diff --git a/termwiz/Cargo.toml b/termwiz/Cargo.toml
index 674c01fd7..c8f0dcd0e 100644
--- a/termwiz/Cargo.toml
+++ b/termwiz/Cargo.toml
@@ -23,7 +23,7 @@ fixedbitset = "0.4"
 fnv = {version="1.0", optional=true}
 hex = "0.4"
 # we want image 0.24.6 or later. see https://github.com/wez/wezterm/issues/3250
-image = {version="0.24", git="https://github.com/image-rs/image.git", rev="fe069785ae245a2c510fd724ef96da283b05a236", optional=true}
+image = { version = "0.24", git = "https://github.com/image-rs/image.git", rev = "fe069785ae245a2c510fd724ef96da283b05a236", optional = true, default-features = false, features = ["gif", "jpeg", "ico", "png", "farbfeld", "jpeg_rayon"] }
 lazy_static = "1.4"
 libc = "0.2"
 log = "0.4"
@@ -96,4 +96,3 @@ harness = false
 [[bench]]
 name = "cell"
 harness = false
-
diff --git a/wezterm-gui/Cargo.toml b/wezterm-gui/Cargo.toml
index b61b1164d..9caa1b429 100644
--- a/wezterm-gui/Cargo.toml
+++ b/wezterm-gui/Cargo.toml
@@ -56,7 +56,7 @@ fuzzy-matcher = "0.3"
 hdrhistogram = "7.1"
 http_req = "0.9"
 # we want image 0.24.6 or later. see https://github.com/wez/wezterm/issues/3250
-image = {version="0.24", git="https://github.com/image-rs/image.git", rev="fe069785ae245a2c510fd724ef96da283b05a236"}
+image = { version = "0.24", git = "https://github.com/image-rs/image.git", rev = "fe069785ae245a2c510fd724ef96da283b05a236", default-features = false, features = ["gif", "jpeg", "ico", "png", "farbfeld", "jpeg_rayon"] }
 lazy_static = "1.4"
 libc = "0.2"
 lfucache = { path = "../lfucache" }

